cli/cmd/inject.go

```diff
@@ -201,6 +212,13 @@ func injectPodTemplateSpec(t *v1.PodTemplateSpec) enhancedPodTemplateSpec {
 		},
 	}
 
+	controlPlaneDNSName := fmt.Sprintf("tcp://proxy-api.%s.svc.cluster.local:%d", controlPlaneNamespace, proxyAPIPort)
+	if t.Namespace == controlPlaneNamespace && containsProxyAPIContainer(t.Spec, "proxy-api") {
```

Comment:
I don't think this is correct. We don't reserve the container name "proxy-api". In the future (if it doesn't already work now) we want to allow people to install the conduit control plane into the same namespace as their app, so we have to assume that there are other objects in the control plane namespace that we don't control. Some of them might have a container named `proxy-api` that is unrelated to our `proxy-api`.

Conversely, we already reserve the name `controller` in the conduit control plane namespace, so I personally think it's sufficient to just check that `t.Name == "controller"`. Regardless, checking `t.Name == "controller"` is at least necessary.